proxy_presentation.php

```diff
@@ -67,16 +67,21 @@
         $bbbinstanceid = substr($bbbinstanceid, 0, strpos($bbbinstanceid, '['));
     }
 
-
     // Check if the user has appropriate access to the meeting resources.
-    $meetingsinstances = $DB->get_records('bigbluebuttonbn', ['meetingid' => $basemeetingid], '', 'id, course');
     $capable = false;
-    foreach ($meetingsinstances as $bbbinstance) {
-        // Checks the BBB instances to see if the user has capabilities for viewing - breaking on first success.
-        if (bigbluebuttonbn_has_capability($bbbinstance->id, 'mod/bigbluebuttonbn:view')) {
-            $capable = true;
-            break;
+    if (!bigbluebuttonbn_has_capability($bbbinstanceid, 'mod/bigbluebuttonbn:view')) {
+        $meetingsinstances = $DB->get_records('bigbluebuttonbn', ['meetingid' => $basemeetingid], '', 'id, course');
+        foreach ($meetingsinstances as $bbbinstance) {
+            // Checks the BBB instances to see if the user has capabilities for viewing - breaking on first success.
+            $course = get_course($bbbinstance->course);
+            if (bigbluebuttonbn_has_capability($bbbinstance->id, 'mod/bigbluebuttonbn:view') && $course->visible) {
```

Comment:
The $course->visible check is good, but I would rather have dead simple checks occur before a function call to something like bigbluebuttonbn_has_capability, so if it fails, _it fails fast_.

Reply:
Updated with the visible check first for short path. Nice catch!